### DISCUSSION TITLE
Removing Authorization headers from logs

### DISCUSSION THREAD

**Discussion by Sachin4403** [xmidt-org/xmidt#52] [General]
We want to skip Authorisation being printed in logs. Anyone who is having access to those logs can harm the system using Authorization and we will never know who did this. Already this is implemented in argus.

https://github.com/xmidt-org/argus/blob/c2562df18c76f64ff59262c55a45bb5804566a00/auth/logger.go#L49-L58

**Comment by joe94** [comment]
Hey @Sachin4403, I agree. Two suggestions I have heard so far are (1) Follow this implementation you show by removing the sensitive headers and (2) Instead of logging all headers, only log those we think we need.

**Reply by Sachin4403** [reply]
Hi @joe94, We can follow both of them for
(1) We just need to log the Authorisation type so that it will help in debugging.
(2) We can configure what all headers should be excluded and this configuration will be coming from properties files.

**Reply by joe94** [reply]
Yeah, I think long-term we could do something along those lines. I think for now as a quick patch, we could just use the fix in Argus.

**Reply by Sachin4403** [reply]
Hi @joe94,

We need to do something here also.
https://github.com/xmidt-org/tr1d1um/blob/main/main.go#L445
https://github.com/xmidt-org/tr1d1um/blob/main/main.go#L450
https://github.com/xmidt-org/talaria/blob/main/primaryHandler.go#L123

Here we are printing the AuthToken in logs.

**Reply by joe94** [reply]
@Sachin4403, those entries could be filtered out in your production environment by setting your log level to something above debug (i.e. `INFO`)

**Reply by Sachin4403** [reply]
Okay, @joe94 will change the log level.

**Reply by joe94** [reply]
@Sachin4403, could you mark this question as "answered"/"resolved"? thanks!

**Reply by Sachin4403** [reply]
> @Sachin4403, could you mark this question as "answered"/"resolved"? thanks!

Hi @joe94, I am not getting an option to mark this as resolved. Can you do it from your end?
